doc/manuals/sprokit/embedded_pipeline.rst

```diff
@@ -0,0 +1,327 @@
+so they can be presented to the pipeline at the same time.
+
+Lines 12 - 14 add individual data values to the pipeline input object
+(adapter_data_set). The string specified in the call must match the
+port name that was used to connect to the input_adapter. The value
+specified will be supplied to that port.
+
+Line 15 sends the set of input data to the input adapter process. An
+error will be thrown if there is a port connected to that process
+which does not have an associated data element. An error will also be
+thrown if there is a element with a name that is not connected to the
+input process.
+
+Line 17 retrieves a set of output values from the pipeline. There will
+be a value for each port that is connected to the output_process.
+
+
+Pipeline Inputs and Outputs
+---------------------------
+
+In order to adapt a pipeline to running in an embedded manner, the
+inputs that are supplied by the program are passed to the
+*input_adapter* process and the outputs from the pipeline are passed to
+the *output_adapter*. The pipeline definition must specify the
+connections from/to these processes.
+
+Sets of input data elements are passed to the pipeline using a
+`adapter_data_set` object. This class defines a names set of data
+items where the name corresponds to the port name, as specified in the
+pipeline definition. The type of the data element must be compatible
+with what is expected on the port by the receiving process. The
+*output_adapter* returns the named data elements in the same way.
+
+..  doxygenclass:: kwiver::adapter::adapter_data_set
+    :project: kwiver
+    :members:
+
+Polling the interface queues
+----------------------------
+
+The above example code uses `send()` and `receive()` in a loop to
+supply data to the pipeline and retrieve the output. While this is a
+direct approach, it will not work if there is any latency in the
+pipeline. That is, if the pipeline will only produce any output after
+some number of inputs are supplied. Both the `send()` and `receive()`
+methods will block if they can not complete, but it is possible to
+check to see if these calls will block or succeed. When ready to call
+`receive()`, the `empty()` method can be called to see if there is an
+*adapter_data_set* available. In the same manner, the `full()` method
+can be called to see if there is space to send a *adapter_data_set*
+before calling `send()`.
+
+
+How to Specify Pipeline
+-----------------------
+
+Pipelines are provided to the *embedded_pipeline* object as a
+stream. The most common types of streams used are file streams and
+string streams. To use a file stream, the controlling program needs to
+open the file and pass the stream to the embedded pipeline
+object. Alternatively, the pipeline can be specified as a string
+stream. The easiest way to build the pipeline definition is to use the
+macros supplied in
+
+``#include <sprokit/pipeline_util/literal_pipeline.h>``
```

Comment:
Will this link to the documentation of that header? (If not, do we have a way to do that?)

Reply:
The controls below will bring in the specific doxy doc for the macros.